modules/nw-controlling-dns-pod-placement.adoc

```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
+//
+// * networking/dns-operator.adoc
+
+[id="nw-controlling-dns-pod-placement_{context}"]
+= Controlling DNS pod placement
+
+The DNS Operator has two daemon sets: one for CoreDNS and one for managing the `/etc/hosts` file. The daemon set for `/etc/hosts` must run on every node host and include an entry for the cluster image registry to support pulling images. Security policies can prohibit communication between pairs of nodes, which prevents the daemon set for CoreDNS from running on every node.
+
+As a cluster administrator, you can use a custom node selector to configure the daemon set for CoreDNS to run or not run on certain nodes.
+
+
+.Prerequisites
+
+* You installed the `oc` CLI.
+* You are logged in to the cluster with a user with `cluster-admin` privileges.
+
+.Procedure
+
+* To prevent communication between certain nodes, configure the `spec.nodePlacement.nodeSelector` API field:
+
+. Modify the DNS Operator object named `default`:
++
+[source, terminal]
+----
+$ oc edit dns.operator/default
+----
++
+. Specify a node selector that includes only control plane nodes in the `spec.nodePlacement.nodeSelector` API field:
++
+[source,yaml]
+----
+ spec:
+   nodePlacement:
+     nodeSelector:
+       node-role.kubernetes.io/worker: ""
+----
+
+* To allow the daemon set for CoreDNS to run on nodes, configure a taint and toleration:
++
+. Modify the DNS Operator object named `default`:
++
+[source,terminal]
+----
+$ oc edit dns.operator/default
+----
++
+. Specify a taint key and a toleration for the taint:
++
+[source,yaml]
+----
+ spec:
+   nodePlacement:
+     tolerations:
+     - effect: NoExecute
+       key: "dns-only"
+       operators: Equal
+       value: abc
+       tolerationSeconds: 3600
```

Comment:
If the taint is "dns-only", then it makes sense to tolerate it indefinitely (i.e., omit `tolerationSeconds`).  However, this is only an example, so maybe it is more important to illustrate all the fields.  (Maybe the value "abc" means "allowed but curtailed".  * grin *.)

Reply:
Ah, makes sense! I'm adding a call out to clarify this point.